src/apis/club-reviews/services/club-reviews.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Inject, Injectable } from '@nestjs/common';
+
+import { ClubReviewDto } from '@src/apis/club-reviews/dto/club-review.dto';
+import { CreateClubReviewDto } from '@src/apis/club-reviews/dto/create-club-review.dto';
+import { ClubReviewRepository } from '@src/apis/club-reviews/repositories/club-review.repository';
+import { IClubReviewRepository } from '@src/apis/club-reviews/repositories/iclub-review.repository';
+import { CLUB_REVIEW_ERROR_CODE } from '@src/constants/error/club-review/club-review-error-code.constant';
+import { HttpConflictException } from '@src/http-exceptions/exceptions/http-conflict.exception';
+
+@Injectable()
+export class ClubReviewsService {
+  constructor(
+    @Inject(ClubReviewRepository)
+    private readonly clubReviewRepository: IClubReviewRepository,
+  ) {}
+
+  async create(
+    createClubReviewDto: CreateClubReviewDto,
+  ): Promise<ClubReviewDto> {
+    const isExistClubReview = await this.clubReviewRepository.isExistClubReview(
+      createClubReviewDto.clubId,
+      createClubReviewDto.userId,
+    );
+
+    if (isExistClubReview) {
+      throw new HttpConflictException({
+        code: CLUB_REVIEW_ERROR_CODE.ALREADY_REVIEWED,
+      });
+    }
+
+    const newClubReview = await this.clubReviewRepository.createClubReview({
+      ...new CreateClubReviewDto(createClubReviewDto),
+    });
```

Comment:
```suggestion
    const newClubReview = await this.clubReviewRepository.createClubReview(new CreateClubReviewDto(createClubReviewDto));
```
클래스로 생성한 인스턴스를 지역변수로 가지고 있지 않으니 이런 경우에는 새로운 객체로 만들지 않아도 됩니다.
스프레드 연산자로 새롭게 만들면 클래스 인스턴스가 아닌 플레인 인스턴스로 변환되기도 하고요

Reply:
플레인 인스턴스로 변환되는 부분은 생각을 못했는데 그렇겠네요!
근데 처음에 말하신 지역 변수는 어떤 말씀인지 잘 모르겠습니다 ㅠㅠ 
```typescript
const clubReviewDto = new CreateClubReviewDto(createClubReviewDto);

const newClubReview = await this.clubReviewRepository.createClubReview(clubReviewDto);
``` 
이 코드와
```typescript
const newClubReview = await this.clubReviewRepository.createClubReview(new CreateClubReviewDto(createClubReviewDto));
``` 
이 코드에서의 차이가 있다는 말씀이실까요?

Reply:
1번 코드를 보면 스코프 내에서 clubReviewDto 라는 변수를 가지고 있고 2번코드는 클래스 인스턴스를 특정 변수에 가지고 있지 않는 상태입니다.
참조타입을 새로운 객체로 넘기는 이유가 순수함수(부수효과가 없는)임이 보장되지 않을 수 있기 때문에 새로운 객체를 넘겨서 내부에서 가지고 있는 변수의 변경을 외부에서 처리하지 않게하기 위함입니다.

Reply:
디테일한 설명 진심으로 감사드립니다!!